### ISSUE TITLE
I got very poor result for Yolov4-mish custom model after training all steps

### ISSUE THREAD

**Issue by akashAD98** [AlexeyAB/darknet#7678]
i trained my model on tesla v100 for 58,000 steps & my Average loss is 23. & My MAP is 43.42%. & even my final.weight & best.weight is not saved after training is completed.

 * what command do you use?
`darknet.exe detector train data/obj.data yolo-obj.cfg yolov4.conv.137 -map`
   
 * what dataset do you use?   
    I have a custom dataset for 29 classes , each class has 2kbounding boxes.. even I tried counter_per_class for handling imbalanced data 
    * what Loss and mAP did you get?
    I'm getting avg loss of 23 & MAP:43%

    * show chart.png with Loss and mAP   
   

Even map is not showing in graph.png so I used this command to calucalte MAP
`To check accuracy mAP@IoU=50: darknet.exe detector map data/obj.data yolo-obj.cfg backup\yolo-obj_7000.weights`

![AP_Electronics](https://user-images.githubusercontent.com/62583018/117124466-377da400-ad4d-11eb-8a4e-2dcc641331e9.png)

my cfg file:(I have modified few parameters according to alexyab mentioned )
[yolov4x-mish_custom_elec_29.cfg.txt](https://github.com/AlexeyAB/darknet/files/6426304/yolov4x-mish_custom_elec_29.cfg.txt)

please let me know what mistake I have done? & how can I solve this error? @AlexeyAB @WongKinYiu @cenit

**Comment by SpongeBab** [comment]
You trained 58000 steps,why you use yolo-obj_7000.weights?
The AP50 in YOLOv4 paper is 43.5,and AlexyAB trained 505000 iterations for 80 classes.You trained 58000 iterations for29 classes, I think it is a good result~
And your avg-loss is 23?23%?or2.3%?  If you get 23% avg-loss,it is no doubt that you should continue to train. Until the loss below 3% around.

**Comment by SpongeBab** [comment]
> > The AP50 in YOLOv4 paper is 43.5%
> 
> No, it's AP50:95, not AP50, AP50=43.53% is not good, maybe your dataset is too difficult.

Oh，sorry，I wrote it wrong.

**Comment by akashAD98** [comment]
> You trained 58000 steps,why you use yolo-obj_7000.weights?
> The AP50 in YOLOv4 paper is 65.7%, and AlexyAB trained 505000 iterations for 80 classes.
> And your avg-loss is 23?23%?or2.3%? If you get 23% avg-loss,it is no doubt that you should continue to train. Until the loss below 3% around.

hey @SpongeBab this MAP is just code that I copied from the readme.I took the last weight for calculating MAP.
i have 29 classes  29*2000=58,000 so I trained for this many steps. 
can I increase the steps  ?? but i think 58k is enough according to alexyAB.

**Comment by akashAD98** [comment]
> > The AP50 in YOLOv4 paper is 43.5%
> 
> No, it's AP50:95, not AP50, AP50=43.53% is not good, maybe your dataset is too difficult.

yes i have 29 classes & each class has almost 2000 images. Even i was doing experiments with different parameters. It will be a great help if you tell me how can I detect both small & big objects. Do you have your own.cfg ??  i tried all yolov4-mish,yolov4-csp,& yolov4-custom. I'm still doing research to find the best parameters. Thanks in dvance

**Comment by SpongeBab** [comment]
@akashAD98 
Whether stop or continue to train,depends on your loss.You can see from there:https://github.com/AlexeyAB/darknet#when-should-i-stop-training
For improvement,you can see: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

**Comment by akashAD98** [comment]
This is another cfg experiment & I got a MAP of 94% & Avg loss is  68 which high. should I again need to train for more steps ?? @AlexeyAB @SpongeBab  My training is good or something is wrong in this?

![WhatsApp Image 2021-05-09 at 4 25 56 AM](https://user-images.githubusercontent.com/62583018/117575260-ebf13000-b095-11eb-8f6a-184c81160bf5.jpeg)

**Comment by ekesdf** [comment]
From an evaluation standpoint your training is going well the loss can'be high even you get a high mAP value bc of your dataset and how "complex" it is. When you have many small and many big or very different things like the one has the size of an ant and the other one has a size of an elephant, you have much noise in your background, or the "contrast/difference" of your labels compared to your background. Like wearing camouflage in the jungle, nobody can see you.

So I would recommend using the current best model and see if those predictions made by the model are enough for. Else train it till the improvement of your loss is very low. As an example it took 1000 epochs to change your loss by like 0.1

**Comment by SpongeBab** [comment]
@akashAD98 oh,so odd.Your map is high and loss is high,too.Usually this is impossible.
If you are sure your cfg file,data file and you doesn't make no change on code.I suggest you continue to train and observe is there any change.

**Comment by akashAD98** [comment]
@SpongeBab I have changed the cfg.I have added few parameters in my cfg and it's increasing my MAP as compared to previous or default  mish-cfg .

**Comment by banyungong1** [comment]
> @SpongeBab我已经改变了cfg。我已经在我的cfg中添加了几个参数，并且与之前或默认的mish-cfg相比，它正在增加我的MAP。
您好我想请教一下，您在您的cfg中添加了哪几个参数，让它增加了您的map
